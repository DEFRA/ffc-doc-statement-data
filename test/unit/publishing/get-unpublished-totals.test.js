const db = require('../../../app/data')
const getUnpublishedTotals = require('../../../app/publishing/total/get-unpublished-total')
const { mockTotal1, mockTotal2, mockTotal3 } = require('../../mocks/totals')

db.total = {
  findAll: jest.fn()
}

describe('getUnpublishedTotals', () => {
  beforeEach(() => {
    db.total.findAll.mockResolvedValue([mockTotal1, mockTotal2, mockTotal3])
  })

  test('getUnpublishedTotals returns the correct data', async () => {
    const transaction = {}
    const result = await getUnpublishedTotals(transaction)
    expect(result).toEqual([mockTotal1, mockTotal2, mockTotal3])

    expect(db.total.findAll).toMatchSnapshot()
  })
})
