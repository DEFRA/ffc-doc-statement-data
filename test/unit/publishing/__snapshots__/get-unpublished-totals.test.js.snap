// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getUnpublishedTotals getUnpublishedTotals returns the correct data 1`] = `
[MockFunction] {
  "calls": [
    [
      {
        "attributes": [
          "calculationId",
          [
            "calculationId",
            "calculationReference",
          ],
          [
            "calculationId",
            "totalsId",
          ],
          "sbi",
          "frn",
          "agreementNumber",
          "claimId",
          [
            "claimId",
            "claimReference",
          ],
          "schemeType",
          "calculationDate",
          "invoiceNumber",
          "agreementStart",
          "agreementEnd",
          "totalAdditionalPayments",
          "totalActionPayments",
          "totalPayments",
          "updated",
          "datePublished",
        ],
        "lock": true,
        "raw": true,
        "skipLocked": true,
        "transaction": {},
        "where": {
          Symbol(or): [
            {
              "datePublished": null,
            },
            {
              "datePublished": {
                Symbol(lt): Col {
                  "col": "updated",
                },
              },
            },
          ],
        },
      },
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": Promise {},
    },
  ],
}
`;
